packages/stacking/tests/utils.test.ts

```diff
@@ -0,0 +1,79 @@
+  },
+  {
+    type: 'p2wpkh',
+    address: 'bc1qzepy04hjksj6c4m3ggawdjqvw48hzu4sy24ghc',
+    expectedHash: '164247d6f2b425ac5771423ae6c80c754f7172b0',
+    expectedVersion: 4,
+    expectedLength: 20,
+    mockedResult: `0x0a0c000000041266697273742d7265776172642d6379636c650100000000000000000000000000001a490b6c6f636b2d706572696f64010000000000000000000000000000000208706f782d616464720c00000002096861736862797465730200000014164247d6f2b425ac5771423ae6c80c754f7172b00776657273696f6e020000000104127265776172642d7365742d696e64657865730b0000000201000000000000000000000000000000000100000000000000000000000000000000`,
+  },
+  {
+    type: 'p2wsh',
+    address: 'bc1qczxqnu5hx2zvxcqt3lmr6vjju4ysf7d649mvrzd8v7l3jez0dqzqgz0ewt',
+    expectedHash: 'c08c09f2973284c3600b8ff63d3252e54904f9baa976c189a767bf19644f6804',
+    expectedVersion: 5,
+    expectedLength: 32,
+    mockedResult: `0x0a0c000000041266697273742d7265776172642d6379636c650100000000000000000000000000001a9c0b6c6f636b2d706572696f64010000000000000000000000000000000208706f782d616464720c00000002096861736862797465730200000020c08c09f2973284c3600b8ff63d3252e54904f9baa976c189a767bf19644f68040776657273696f6e020000000105127265776172642d7365742d696e64657865730b0000000201000000000000000000000000000000000100000000000000000000000000000000`,
+  },
+  {
+    type: 'p2tr',
+    address: 'bc1p8dlmzllfah294ntwatr8j5uuvcj7yg0dete94ck2krrk0ka2c9qqwwn4dr',
+    expectedHash: '3b7fb17fe9edd45acd6eeac679539c6625e221edcaf25ae2cab0c767dbaac140',
+    expectedVersion: 6,
+    expectedLength: 32,
+    mockedResult: `0x0a0c000000041266697273742d7265776172642d6379636c650100000000000000000000000000001a9f0b6c6f636b2d706572696f64010000000000000000000000000000000208706f782d616464720c000000020968617368627974657302000000203b7fb17fe9edd45acd6eeac679539c6625e221edcaf25ae2cab0c767dbaac1400776657273696f6e020000000106127265776172642d7365742d696e64657865730b0000000201000000000000000000000000000000000100000000000000000000000000000000`,
+  },
+];
+
+test.each(BTC_ADDRESS_CASES)(
+  'parsing btc address $type %#',
+  ({ address, expectedVersion, expectedHash, expectedLength }) => {
+    const decoded = decodeBtcAddress(address);
+    expect(decoded.version).toBe(expectedVersion);
+    expect(decoded.data).toEqual(hexToBytes(expectedHash));
+    expect(decoded.data).toHaveLength(expectedLength);
+
+    const tuple = poxAddressToTuple(address);
+    expect(tuple.data['version'].buffer).toHaveLength(1);
+    expect(tuple.data['version'].buffer[0]).toBe(expectedVersion);
+    expect(tuple.data['hashbytes'].buffer).toEqual(hexToBytes(expectedHash));
+    expect(tuple.data['hashbytes'].buffer).toHaveLength(expectedLength);
+  }
+);
```

Comment:
Could you please add additional test vectors here for more address type permutations? For example, could you take tests/vectors from here: https://github.com/hirosystems/stacks-blockchain-api/blob/ae0eb65c4d6901db172f2b4ea751817d8ef8c05c/src/tests/helpers-tests.ts#L193-L530

Reply:
I've added a decode/encode test suite with those test vectors. Do you think it should also duplicate some of the tests you linked here?

Also, do you know how I could setup tests with vectors closer to the `hashbytes` on the blockchain — i.e. compare to hashbytes that I know will be received when paying out through blockchain/miner code. I'm trying to sift through the codebase (`reward_address.bytes()`, `PoxAddress`, etc.), but haven't found any useful tests...

Reply:
Thanks! I think there still might be missing vectors from the API repo, for example the P2TR ones. Or am I just missing them?

Reply:
> Also, do you know how I could setup tests with vectors closer to the hashbytes on the blockchain — i.e. compare to hashbytes that I know will be received when paying out through blockchain/miner code. I'm trying to sift through the codebase (reward_address.bytes(), PoxAddress, etc.), but haven't found any useful tests...

Not really. I think that would require more of an end-to-end test. Perhaps you could grab those outputs manually to use as test vectors?

Reply:
> Thanks! I think there still might be missing vectors from the API repo, for example the P2TR ones. Or am I just missing them?

You're completely right! 😬 Added now

Reply:
Added a bunch more test vectors.
This was the only one I couldn't reproduce: https://gist.github.com/janniks/b7ba98b1770228880ef05fc96ccd14cb#file-btc-address-test-vectors-js-L346-L354